Assets/Scripts/Event Manager.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+
+[CreateAssetMenu(fileName = "EventManager", menuName = "Scriptable Objects/EventManager")]
+public class EventManager : ScriptableObject
```

Comment:
This is a very quick implementation of a singleton but when used with scriptable objects can sometimes perform strangely. 

I have used the following before a lot as a base class to inherit from to make sure I get it right: 

https://gist.github.com/megonemad1/5789fa9783b46c8ab9ed70512f63e4b5

Its worth looking at that class and understanding why its set up the way it is. You wouldn't need to call Initialize then, and not need to re-implement the singleton patter for all of your scriptableobject singletons. I believe unity also has an inbuild implementation of this, also called ScriptableSingleton, so its worth looking into that as well, but the link I provided is nicer because it lets you see how its working under the hood (and debug etc)

Reply:
EventManager : ScriptableSingleton<EventManager>

Reply:
Resolved